### PR TITLE
[BugFix] Fix bug existing partitions send unnecessary requests to BE

### LINKED ISSUE #18162: Alter table if not exists is executed multiple times, and the existing partitions will still send unnecessary requests to BE
### Steps to reproduce the behavior (Required)
1. CREATE TABLE site_access(
    event_day datetime,
    site_id INT DEFAULT '10',
    city_code VARCHAR(100),
    user_name VARCHAR(32) DEFAULT '',
    pv BIGINT DEFAULT '0'
)
DUPLICATE KEY(event_day, site_id, city_code, user_name)
PARTITION BY time_slice(event_day, 5, 'day', 'floor') (
START ("2015-01-01") END ("2022-01-01") EVERY (INTERVAL 1 year)
)
DISTRIBUTED BY HASH(event_day, site_id) BUCKETS 32
PROPERTIES("replication_num" = "1");
2. alter table site_access add PARTITION if not exists p202301 VALUES [("2023-01-01"), ("2023-01-03"));

### Expected behavior (Required)
be does not show create tablet log
### Real behavior (Required)
be has create tablet log
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
main,3.0

### PR DESCRIPTION
**Pull request by Astralidea**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #18162

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by Astralidea** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 1]
```diff
@@ -1089,15 +1089,19 @@ private void checkDataProperty(List<PartitionDesc> partitionDescs) {
         }
     }
 
-    private List<Partition> createPartitionList(Database db, OlapTable copiedTable, List<PartitionDesc> partitionDescs,
-                                                HashMap<String, Set<Long>> partitionNameToTabletSet,
-                                                Set<Long> tabletIdSetForAll)
+    private Map<Partition, PartitionDesc> createPartitionMap(Database db, OlapTable copiedTable,
+                                                             List<PartitionDesc> partitionDescs,
+                                                             HashMap<String, Set<Long>> partitionNameToTabletSet,
+                                                             Set<Long> tabletIdSetForAll, Set<String> existPartitionName)
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:1095]:
```suggestion
                                                             Set<Long> tabletIdSetForAll, Set<String> existPartitionNameSet)
```

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 21 / 21 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/ListPartitionInfo.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/RangePartitionInfo.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|14|14|100.00%|[]|

**Comment by Astralidea** [comment]
run all

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 21 / 21 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/ListPartitionInfo.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/RangePartitionInfo.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|14|14|100.00%|[]|

**Review by gengjun-git — APPROVED** [review]

**Review by ABingHuang — APPROVED** [review]
LGTM

**Comment by Astralidea** [comment]
run starrocks_admit_test

**Comment by Astralidea** [comment]
run starrocks_admit_test

**Comment by wangruin** [comment]
run starrocks_admit_test

**Comment by Astralidea** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by Astralidea** [comment]
https://github.com/Mergifyio backport branch-3.0

**Pull request merged**